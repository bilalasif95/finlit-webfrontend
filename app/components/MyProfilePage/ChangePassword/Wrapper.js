import styled from 'styled-components';

const Wrapper = styled.div`
  .basic_info {
    height: 100%;
    .form-group {
      margin: 0 0 2rem;
    }
    .form_footer {
      // position: absolute;
      // bottom: 50px;
      // right: 40px;
    }
  }
`;

export default Wrapper;
