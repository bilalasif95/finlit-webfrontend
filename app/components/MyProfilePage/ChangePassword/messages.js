/*
 * Webinars Filters Messages
 *
 * This contains all the text for the Webinars Filters component.
 */
import { defineMessages } from 'react-intl';

export const scope = 'boilerplate.components.WebinarsFilters';

export default defineMessages({
  ChangePassword: {
    id: `${scope}.ChangePassword`,
    defaultMessage: 'Change Password',
  },
  YourCurrentPassword: {
    id: `${scope}.YourCurrentPassword`,
    defaultMessage: 'Enter current password',
  },
  NewPassword: {
    id: `${scope}.NewPassword`,
    defaultMessage: 'Enter new password',
  },
  ConfirmPassword: {
    id: `${scope}.ConfirmPassword`,
    defaultMessage: 'Confirm new password',
  },
  Cancel: {
    id: `${scope}.Cancel`,
    defaultMessage: 'Cancel',
  },
  Save: {
    id: `${scope}.Save`,
    defaultMessage: 'Save',
  },
});
