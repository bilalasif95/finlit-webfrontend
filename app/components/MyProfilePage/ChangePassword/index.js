/*
 * Change Password Component
 */
import React, { useEffect, useState } from 'react';
import { FormattedMessage } from 'react-intl';
import {
  FormGroup,
  Label,
  Row,
  Col,
  Button,
  Input,
  InputGroup,
  InputGroupAddon,
  // FormText,
} from 'reactstrap';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import InputBase from '@material-ui/core/InputBase';
import { withStyles } from '@material-ui/core/styles';
import Switch from '@material-ui/core/Switch';
import { BsPencil } from 'react-icons/bs';
import { FaCheckCircle } from 'react-icons/fa';
import axios from 'axios';
import QRCode from 'qrcode.react';
import { FiCamera } from 'react-icons/fi';
import messages from './messages';
import Wrapper from './Wrapper';
import { API } from '../../../config/config';

function ChangePassword() {
  return (
    <Wrapper>
      <div className="basic_info">
        <h4>
          <FormattedMessage {...messages.ChangePassword} />
        </h4>
        <Row>
          <Col lg={6} md={6} sm={6} xs={12}>
            <FormGroup>
              {/* <Label for="password">{apiError || ''}</Label>
              <Label for="password">{apiSuccessMessage || ''}</Label> */}
              <Label for="password">
                <FormattedMessage {...messages.YourCurrentPassword} />
              </Label>
              <Input
                type="password"
                name="currentPassword"
                id="password"
                placeholder="**********"
              // onChange={e => handleChangeEvent(e)}
              />
              {/* <Label for="password">
                {errors.currentPassword ? errors.currentPassword : ''}
              </Label> */}
            </FormGroup>


          </Col>
          <Col lg={6} md={6} sm={6} xs={12}>
            <FormGroup>
              <Label for="password">
                <FormattedMessage {...messages.NewPassword} />
              </Label>
              <Input
                type="password"
                name="newPassword"
                id="password"
                placeholder="**********"
              // onChange={e => handleChangeEvent(e)}
              />
              {/* <Label for="password">
                {errors.newPassword ? errors.newPassword : ''}
              </Label> */}
            </FormGroup>
          </Col>
          <Col lg={6} md={6} sm={6} xs={12}>
            <FormGroup>
              <Label for="password">
                <FormattedMessage {...messages.ConfirmPassword} />
              </Label>
              <Input
                type="password"
                name="confirmPassword"
                id="password"
                placeholder="**********"
                onChange={e => handleChangeEvent(e)}
              />
              {/* <Label for="password">
                {errors.confirmPassword ? errors.confirmPassword : ''}
              </Label>
              <Label for="password">
                {errors.passwordMatching ? errors.passwordMatching : ''}
              </Label> */}
            </FormGroup>
          </Col>
        </Row>
        <div className="form_footer">
          <div className="bottom_btns">
            <Button className="btn_save">
              <FormattedMessage {...messages.Cancel} />
            </Button>
            <Button
              className="btn_submit"
            // onClick={handleUpdateProfileSave}
            >
              <FormattedMessage {...messages.Save} />
            </Button>
          </div>
        </div>
      </div>
    </Wrapper>
  );
}

export default ChangePassword;
