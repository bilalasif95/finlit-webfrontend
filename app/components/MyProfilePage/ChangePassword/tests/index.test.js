import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory, BrowserRouter } from 'react-router-dom';

import configureStore from '../../../../configureStore';
import ChangePassword from '../index';

describe('<ChangePassword />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <Provider store={store}>
          <IntlProvider locale="en">
            <BrowserRouter>
              <ChangePassword />
            </BrowserRouter>
          </IntlProvider>
        </Provider>,
      )
      .toJSON();

    expect(renderedComponent).toMatchSnapshot();
  });
});
