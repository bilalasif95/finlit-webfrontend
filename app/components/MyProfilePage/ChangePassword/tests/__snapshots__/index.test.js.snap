// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BootCampFilters /> should render and match the snapshot 1`] = `
<div
  className="Wrapper-c1bq11-0 gHkJZl"
  id="filters"
>
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <div
        className="course_filters"
      >
        <div
          className="form-group"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              className=""
            >
              <span>
                Courses
              </span>
            </label>
            <div
              className="MuiInputBase-root WithStyles(ForwardRef(InputBase))-root-1 MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input WithStyles(ForwardRef(InputBase))-input-2"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                English
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput"
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value="0"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="form-group"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              className=""
            >
              <span>
                Courses Type
              </span>
            </label>
            <div
              className="MuiInputBase-root WithStyles(ForwardRef(InputBase))-root-1 MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input WithStyles(ForwardRef(InputBase))-input-2"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                High
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput"
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value="0"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="form-group"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              className=""
            >
              <span>
                Language
              </span>
            </label>
            <div
              className="MuiInputBase-root WithStyles(ForwardRef(InputBase))-root-1 MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input WithStyles(ForwardRef(InputBase))-input-2"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                English
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput"
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value="0"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="form-group"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth"
          >
            <label
              className=""
            >
              <span>
                Level
              </span>
            </label>
            <div
              className="MuiInputBase-root WithStyles(ForwardRef(InputBase))-root-1 MuiInputBase-fullWidth MuiInputBase-formControl"
              onClick={[Function]}
            >
              <div
                aria-haspopup="listbox"
                className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input WithStyles(ForwardRef(InputBase))-input-2"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                Level 1
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput"
                onAnimationStart={[Function]}
                onChange={[Function]}
                required={false}
                tabIndex={-1}
                value="0"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSelect-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-lg-12"
    >
      <div
        className="total_results"
      >
        Showing 200 total results for BootCamp
      </div>
    </div>
  </div>
</div>
`;
